sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -455,7 +457,7 @@ public final class StorageBlobClient: PipelineClient {
         file sourceUrl: URL,
         toContainer container: String,
         asBlob blob: String,
-        properties: BlobProperties? = nil,
+        properties: BlobProperties,
```

Comment:
We should find a way to let this continue to be optional. I don't believe that there's any `BlobProperties` member that is -required- to be set by the customer.

Reply:
I will do this in a follow-up PR and thus left the TODOs in place. I don't want to block Mania's work while I investigate the best path forward, and converting this to a class caused all kinds of errors in XCode, so this was the most straightforward solution.